.github/workflows/static.yml

```diff
@@ -0,0 +1,34 @@
+name: Static and unit checks
+
+on: pull_request
+
+jobs:
+  static:
+    runs-on: ubuntu-latest
+    name: "Running ${{ matrix.test }} tests with perl v${{ matrix.perl-version }}"
+    strategy:
+      matrix:
+        include:
+          - test: static
+            perl-version: 5.26
+          - test: unit
+            perl-version: 5.26
+    container:
+      image: perldocker/perl-tester:${{ matrix.perl-version }}
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install dependencies
+      run: apt-get -y install yamllint libdbus-1-dev libssh2-1-dev parallel
+    - name: Setup perl
+      run: |
+        echo "requires 'Code::DRY';" >> cpanfile
+        echo "requires 'Regexp::Common';" >> cpanfile
+        echo "requires 'Perl::Tidy', '== 20201001';" >> cpanfile
+        make prepare
+    - name: Run ${{ matrix.test }} tests
+      env:
+        TESTS: ${{ matrix.test }}
+      run: |
+        git checkout cpanfile
+        git fetch origin master
+        make test
```

Comment:
I consider it easier to test and also easier to apply for different CIs when you can extract the script sections into individual scripts. Compare to https://github.com/os-autoinst/os-autoinst/blob/master/.travis.yml#L6 , only 6 lines in that file :)

Reply:
Let me make it works as it was in travis before we start with improvements ;)

Reply:
Also, do not forget that we need some budget for it too, as 2000 minutes per 20 days doesn't seem to be enough for us, even if we improve CI to run faster (now due to compile check for all files it takes 10+ minutes).
So I just want to make first step in case we decide to switch.

Reply:
yeah sure. What I actually meant is to first *refactor* what we have in travis CI, then have simple files which can be called in parallel in both github actions and travis CI so that we can also test and see which one is faster (or which one exceeds free quota first) and then drop one

Reply:
Please, check PR description, github actions are for free for public repos. And main issue is not that it takes long to run (even though we can improve there), but that jobs now are picked up very late, e.g. here last push was 3 hours ago: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11551
And I had cases when after 10 hours CI still would not run. That's why I decided to act on this now as it really affects our daily work.

Reply:
Of course. No doubt about that. I appreciate that you prepare a switch to github actions. By the way, I am also doing this: https://github.com/os-autoinst/os-autoinst/pull/1576 :)